sources/Inputs/src/HvdcLine.cpp

```diff
@@ -0,0 +1,40 @@
+//
+// Copyright (c) 2020, RTE (http://www.rte-france.com)
+// See AUTHORS.txt
+// All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at http://mozilla.org/MPL/2.0/.
+// SPDX-License-Identifier: MPL-2.0
+//
+
+/**
+ * @file  HvdcLine.cpp
+ *
+ * @brief Hvdc line structure implementation file
+ *
+ */
+
+#include "HvdcLine.h"
+namespace dfl {
+namespace inputs {
+HvdcLine::HvdcLine(const std::string& id, const ConverterType converterType, const ConverterId& converter1_id, const BusId& converter1_busId,
+                   const boost::optional<bool>& converter1_voltageRegulationOn, const ConverterId& converter2_id, const BusId& converter2_busId,
+                   const boost::optional<bool>& converter2_voltageRegulationOn) :
+    id{id},
+    converterType{converterType},
+    converter1_id{converter1_id},
+    converter1_busId{converter1_busId},
+    converter1_voltageRegulationOn{converter1_voltageRegulationOn},
+    converter2_id{converter2_id},
+    converter2_busId{converter2_busId},
+    converter2_voltageRegulationOn{converter2_voltageRegulationOn} {}
+
+bool
+HvdcLine::operator==(const HvdcLine& other) const {
```

Comment:
To be consistent with the further operators for algorithm definitions, define the equality operator only using the id

Reply:
Even if it is not used ?

Reply:
if it is not used, remove it

Reply:
It is used in the tests, and it needs to test every member to be exhaustive. Maybe you want me to rename the function ?

Reply:
If you need in the tests, implement this operator or equivalent only in tests